### PR TITLE
Planner cleaning: cleanup and refactor

### PR DESCRIPTION
**Pull request by systay**
## Description
While working on #16427, I've stumbled across quite a few annoyances in the code that needed fixing before we could finish the CTE work. Instead of making that PR even bigger, I'm extracting all the clean ups into this PR.

The kind of clean ups are:
 * Use `AddWSColumn` instead of `AddColumn` to add `weight_strings`.
 * Don't push down literals - just evaluate them in the vtgate

## Checklist

-   [ ] "Backport to:" labels have been added if this change should be back-ported to release branches
-   [ ] If this change is to be back-ported to previous releases, a justification is included in the PR description
-   [ ] Tests were added or are not required
-   [ ] Did the new or modified tests pass consistently locally and on CI?
-   [ ] Documentation was added or is not required

### CODE REVIEW

**Review by harshit-gangal — APPROVED** [review]

File: go/vt/vtgate/planbuilder/operators/projection_pushing.go [hunk 1]
```diff
@@ -334,23 +334,50 @@ func splitUnexploredExpression(
...
+
+	// Update the left and right child columns and names based on the applyJoinColumn type.
+	switch {
+	case col.IsPureLeft():
+		lhs.add(pe, alias)
+		col.LHSExprs[0].Expr = colName
+	case col.IsPureRight():
+		rhs.add(pe, alias)
+		col.RHSExpr = colName
+	case col.IsMixedLeftAndRight():
```
Comment (GuptaManan100) [inline go/vt/vtgate/planbuilder/operators/projection_pushing.go:364]: can this just be the default? If the column is not solved by only left or right, then it has to be mixed, right? We already panic in getJoinColumnFor if that is not the case

File: go/vt/vtgate/planbuilder/operators/projection_pushing.go [hunk 2]
```diff
@@ -334,23 +334,50 @@ func splitUnexploredExpression(
...
+		for _, lhsExpr := range col.LHSExprs {
+			ae := aeWrap(lhsExpr.Expr)
+			columnName := ae.ColumnName()
+			ae.As = sqlparser.NewIdentifierCI(columnName)
```
Comment (GuptaManan100) [inline go/vt/vtgate/planbuilder/operators/projection_pushing.go:368]: Why is this required? Isn't this redundant? As in the column name would still be this without the alias, right?
